File: simulation_prototype/Menu.py
import json

from Pizza import Pizza
from Drink import Drink

class Menu:
    def __init__(self):
        self._productList  = []
  
        self._creatMenu()


    def getProductList(self):
        return self._productList


    def getProductByID(self, productID):
        return self._productList[productID]


    def _creatMenu(self):
        with open("products_data.json") as file:
            textMenu = json.load(file)

        for product in textMenu:
            if (textMenu[product]["type"] == "Pizza"):
                new_product = Pizza(len(self._productList), textMenu[product]["name"], int(textMenu[product]["price"]), int(textMenu[product]["eatingTime"]), int(textMenu[product]["prepareTime"]))
            elif (textMenu[product]["type"] == "Drink"):
                new_product = Drink(len(self._productList), textMenu[product]["name"], int(textMenu[product]["price"]), int(textMenu[product]["drinkingTime"]))

            self._productList.append(new_product)

File: Menu.cpp
#include "Menu.h"
// #include "Product.h"
// #include "Pizza.h"
// #include "Drink.h"
// #include "Dish.h"

// #include "json/json.hpp"

// #include <iostream>
// #include <fstream>
// #include <string>

Menu::Menu(std::vector<Product*> menuProductsList)
{
    for (auto product_ptr : menuProductsList)
    {
        productList.push_back(product_ptr);
    }
    // createMenu(products_data_file_name);
}

// void Menu::createMenu(std::string products_data_file_name)
// {
//     using json = nlohmann::json;
//     std::ifstream filestream(products_data_file_name);
//     json jsonFile;
//     filestream >> jsonFile;
    
//     Product *newProduct;
//     for (json::iterator it = jsonFile.begin(); it != jsonFile.end(); ++it) 
//     {
//         if((*it)["type"] == "Pizza")
//         {
//             newProduct = new Pizza(productList.size(), (*it)["name"], std::stoi(std::string((*it)["price"])), std::stoi(std::string((*it)["eatingTime"])), std::stoi(std::string((*it)["prepareTime"])));
//         }
//         else if((*it)["type"] == "Drink")
//         {
//             newProduct = new Drink(productList.size(), (*it)["name"], std::stoi(std::string((*it)["price"])), std::stoi(std::string((*it)["drinkingTime"])));
//         }
//         else if((*it)["type"] == "Dish")
//         {
//             newProduct = new Dish(productList.size(), (*it)["name"], std::stoi(std::string((*it)["price"])), std::stoi(std::string((*it)["eatingTime"])), std::stoi(std::string((*it)["prepareTime"])));
//         }
        
//         productList.push_back(newProduct);
//     }
// }

std::vector<Product*> Menu::getProductList()
{
    return productList;
}

Menu::~Menu()
{
    for(auto product : productList)
    {
        delete product;
    }
}